tests/libs/common/common_tests.cpp

```diff
@@ -115,21 +115,21 @@ ring_buffer_test_event_handler(_In_ void* ctx, _In_opt_ void* data, size_t size)
 {
     ring_buffer_test_event_context_t* event_context = reinterpret_cast<ring_buffer_test_event_context_t*>(ctx);
 
-    if (event_context->canceled) {
-        // Ring buffer subscription is canceled.
-        // The test handler expects this callback due to unsubscribe call.
+    if ((data == nullptr) || (size == 0)) {
+        // eBPF ring buffer will invoke this callback with NULL data indicating the subscription is canceled.
         // Free the callback context and return error so that no further callback is made.
         delete event_context;
-        return -1;
```

Comment:
Line 120 says "return error" but this is changed to return success, is it not?
I suspect the comment in line 120 needs to be updated now, as the error case is line 127.